### PR TITLE
#2760 readme update

### PR DESCRIPTION
**Pull request by NateWr**
Probably easiest to see this in action on my fork: https://github.com/NateWr/wp-cli/tree/i2760_readme_update

Should tick all the boxes in #2760 except for installation requirements (see https://github.com/wp-cli/wp-cli/issues/2760#issuecomment-220031625).

Happy to go in a different direction with anything you don't like the look of.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,91 @@
-WP-CLI
-======
+# WP-CLI
```
Comment (danielbachhuber) [inline README.md]: Let's keep this as:

```
WP-CLI
======
```

It will be easier to parse this out to turn it into the index.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,91 @@
...
+
+Command-line tools for managing WordPress installations. [http://wp-cli.org/](http://wp-cli.org/).
```
Comment (danielbachhuber) [inline README.md]: What about (from the existing homepage):

> [WP-CLI](http://wp-cli.org/) is a set of command-line tools for managing WordPress installations. You can update plugins, set up multisite installs and much more, without using a web browser.

File: README.md [hunk 3]
```diff
@@ -1,32 +1,91 @@
...
 
 [![Build Status](https://travis-ci.org/wp-cli/wp-cli.png?branch=master)](https://travis-ci.org/wp-cli/wp-cli)
 
-WP-CLI is a set of command-line tools for managing WordPress installations.
+Quick links: [Using](#using) | [Installing](#installing) | [Support](#support) | [Extending](#extending) | [Contributing](#contributing) | [Credits](#credits)
+
+## Using
+
+This project aims to provide command-line support for any action you can perform in the WordPress admin and many you can't. So you can [install a plugin](http://wp-cli.org/commands/plugin/install/):
```
Comment (danielbachhuber) [inline README.md]: > The goal of WP-CLI is to provide a command-line interface for any action you can perform in the WordPress admin. The project also includes commands for many actions you can't perform in the WordPress admin.
>
> For instance, `wp plugin install` ([doc](http://wp-cli.org/commands/plugin/install/)) lets you install and activate a WordPress plugin:
>
>     $ wp plugin install rest-api --activate
>     Installing WordPress REST API (Version 2) (2.0-beta13)
>     Downloading install package from https://downloads.wordpress.org/plugin/rest-api.2.0-beta13.zip...
>     Unpacking the package...
>     Installing the plugin...
>     Plugin installed successfully.
>     Activating 'rest-api'...
>     Success: Plugin 'rest-api' activated.
>
> [demonstrate one other command]

File: README.md [hunk 4]
```diff
@@ -1,32 +1,91 @@
...
+$ wp plugin install rest-api
+```
+
+And [activate it](http://wp-cli.org/commands/plugin/activate/):
+
+```
+$ wp plugin activate rest-api
+```
+
+View the [Quick Start](http://wp-cli.org/docs/quick-start/) guide or jump to the [complete list of commands](http://wp-cli.org/commands/) for managing themes and plugins, importing and exporting data, performing database search-replace operations and more.
```
Comment (danielbachhuber) [inline README.md]: > For a complete introduction, view the [Quick Start guide](http://wp-cli.org/docs/quick-start/). If you already feel comfortable with the basics, jump into...

File: README.md [hunk 5]
```diff
@@ -1,32 +1,91 @@
...
+## Support
 
-If you want to receive an email for every single commit, you can subscribe to the [wp-cli-commits](https://groups.google.com/forum/?fromgroups=#!forum/wp-cli-commits) mailing list.
+WP-CLI is a volunteer-led project and can't support every user individually. If you run into trouble, here are some places to look for help:
 
-I'm running into troubles, what can I do?
------------------------------------------
-To suggest a feature, report a bug, or general discussion, visit the [issues section](https://github.com/wp-cli/wp-cli/issues).
+- [Common issues and their fixes](http://wp-cli.org/docs/common-issues/)
+- [Bug reports](http://wp-cli.org/docs/bug-reports/)
```
Comment (danielbachhuber) [inline README.md]: > Submit a bug report

File: README.md [hunk 6]
```diff
@@ -1,32 +1,91 @@
...
+- [External resources](http://wp-cli.org/docs/external-resources/)
 
-If you're reporting a bug, please include the output from `wp --info` and do your best to follow [bug reporting best practices](http://wp-cli.org/docs/bug-reports/).
+## Extending
```
Comment (danielbachhuber) [inline README.md]: Can we include a short summary of what a command is?

File: README.md [hunk 7]
```diff
@@ -1,32 +1,91 @@
...
+
+Creating a custom WP-CLI command can be easier than it looks. Read the [commands cookbook](http://wp-cli.org/docs/commands-cookbook/).
+
+## Contributing
+
+Thanks for helping to improve WP-CLI. Please read about [creating an issue](http://wp-cli.org/docs/bug-reports/) or [submitting a pull request](http://wp-cli.org/docs/pull-requests/).
+
+### Contributors
```
Comment (danielbachhuber) [inline README.md]: Let's call this "Leadership"

File: README.md [hunk 8]
```diff
@@ -1,32 +1,91 @@
...
+
+```
+$ wp plugin activate rest-api
+```
+
+View the [Quick Start](http://wp-cli.org/docs/quick-start/) guide or jump to the [complete list of commands](http://wp-cli.org/commands/) for managing themes and plugins, importing and exporting data, performing database search-replace operations and more.
+
+## Installing
+
+The recommended way to install WP-CLI is to download the Phar build, mark it executable and place it in your PATH.
```
Comment (danielbachhuber) [inline README.md]: > Before you install though, please make sure your environment meets the minimum requirements:
> * UNIX-like environment (OS X, Linux, FreeBSD, Cygwin); limited support in Windows environment
> * PHP 5.3.29 or later
> * WordPress 3.7 or later

File: README.md [hunk 9]
```diff
@@ -1,32 +1,91 @@
...
-Where can I get more info?
---------------------------
-For documentation and examples, check out [wp-cli.org](http://wp-cli.org/) and the [documentation portal](http://wp-cli.org/docs/).
+Now that you've got WP-CLI, read the [Quick Start](http://wp-cli.org/docs/quick-start/) guide. Or view the [full installation guide](http://wp-cli.org/docs/installing/) with alternative installation methods.
 
-Also, WordPress Answers has a growing list of [WP-CLI related questions](http://wordpress.stackexchange.com/questions/tagged/wp-cli).
+## Support
 
-If you want to receive an email for every single commit, you can subscribe to the [wp-cli-commits](https://groups.google.com/forum/?fromgroups=#!forum/wp-cli-commits) mailing list.
+WP-CLI is a volunteer-led project and can't support every user individually. If you run into trouble, here are some places to look for help:
```
Comment (danielbachhuber) [inline README.md]: I'm not quite sure about this language:

> WP-CLI is a volunteer-led project and can't support every user individually.

I do try to support everyone, but maybe we should have some communication of what types of issues are supported and what aren't.

**Comment by danielbachhuber** [comment]
@NateWr Top notch! I've left a first round of comments.

**Comment by NateWr** [comment]
Ok, updated according to your comments. All pretty straightforward except take a look at the language under Support and the second demonstration of a command under Using.

File: README.md [hunk 10]
```diff
@@ -1,32 +1,110 @@
...
+```
+$ wp transient delete-all
+Success: 34 transients deleted from the database.
+```
+
+For a complete introduction, read the [Quick Start guide](http://wp-cli.org/docs/quick-start/). If you already feel comfortable with the basics, jump into the [complete list of commands](http://wp-cli.org/commands/) for managing themes and plugins, importing and exporting data, performing database search-replace operations and more.
+
+## Installing
+
+The recommended way to install WP-CLI is to download the Phar build, mark it executable and place it in your PATH.
```
Comment (danielbachhuber) [inline README.md]: > See also our documentation on [alternative installation methods](http://wp-cli.org/docs/installing/)

File: README.md [hunk 11]
```diff
@@ -1,32 +1,110 @@
...
+php.ini used:   /etc/php5/cli/php.ini
+WP-CLI root dir:        /home/wp-cli/.wp-cli
+WP-CLI packages dir:    /home/wp-cli/.wp-cli/packages/
+WP-CLI global config:   /home/wp-cli/.wp-cli/config.yml
+WP-CLI project config:
+WP-CLI version: 0.23.0
+```
 
-Also, WordPress Answers has a growing list of [WP-CLI related questions](http://wordpress.stackexchange.com/questions/tagged/wp-cli).
+Now that you've got WP-CLI, read the [Quick Start](http://wp-cli.org/docs/quick-start/) guide. Or view the [full installation guide](http://wp-cli.org/docs/installing/) with alternative installation methods.
```
Comment (danielbachhuber) [inline README.md]: If we link to alternative installation methods in the beginning of this section, I don't think we need this paragraph.

File: README.md [hunk 12]
```diff
@@ -1,32 +1,110 @@
...
+
+## Extending
+
+A **command** is an atomic unit of WP-CLI functionality. `wp plugin install` ([doc](http://wp-cli.org/commands/plugin/install/)) is one command. `wp plugin activate` ([doc](http://wp-cli.org/commands/plugin/activate/)) is another.
+
+WP-CLI comes with dozens of commands. But it's easier than it looks to create a custom WP-CLI command. Read the [commands cookbook](http://wp-cli.org/docs/commands-cookbook/) to learn more.
+
+## Contributing
+
+Thanks for helping to improve WP-CLI. Please read about [creating an issue](http://wp-cli.org/docs/bug-reports/) or [submitting a pull request](http://wp-cli.org/docs/pull-requests/).
```
Comment (danielbachhuber) [inline README.md]: > To get involved, please first read about ...

File: README.md [hunk 13]
```diff
@@ -1,32 +1,110 @@
...
+Installing WordPress REST API (Version 2) (2.0-beta13)
+Downloading install package from https://downloads.wordpress.org/plugin/rest-api.2.0-beta13.zip...
+Unpacking the package...
+Installing the plugin...
+Plugin installed successfully.
+Activating 'rest-api'...
+Success: Plugin 'rest-api' activated.
+```
+
+`wp transient` ([doc](http://wp-cli.org/commands/transient/)) lets you delete one or all transients:
```
Comment (danielbachhuber) [inline README.md]: > Similarly, `wp transient` ...

File: README.md [hunk 14]
```diff
@@ -1,32 +1,110 @@
...
-Also, WordPress Answers has a growing list of [WP-CLI related questions](http://wordpress.stackexchange.com/questions/tagged/wp-cli).
+Now that you've got WP-CLI, read the [Quick Start](http://wp-cli.org/docs/quick-start/) guide. Or view the [full installation guide](http://wp-cli.org/docs/installing/) with alternative installation methods.
 
-If you want to receive an email for every single commit, you can subscribe to the [wp-cli-commits](https://groups.google.com/forum/?fromgroups=#!forum/wp-cli-commits) mailing list.
+## Support
 
-I'm running into troubles, what can I do?
------------------------------------------
-To suggest a feature, report a bug, or general discussion, visit the [issues section](https://github.com/wp-cli/wp-cli/issues).
+WP-CLI's project maintainers do their best to respond to all bug reports and configuration errors within reason and the constraints on their time. Before requesting help, please read to find a solution to your problem in the following resources:
```
Comment (danielbachhuber) [inline README.md]: > the constraints on their volunteered time.

File: README.md [hunk 15]
```diff
@@ -1,32 +1,110 @@
...
-If you're reporting a bug, please include the output from `wp --info` and do your best to follow [bug reporting best practices](http://wp-cli.org/docs/bug-reports/).
+- [Common issues and their fixes](http://wp-cli.org/docs/common-issues/)
+- [Submit a bug report](http://wp-cli.org/docs/bug-reports/)
+- [External resources](http://wp-cli.org/docs/external-resources/)
+
+## Extending
+
+A **command** is an atomic unit of WP-CLI functionality. `wp plugin install` ([doc](http://wp-cli.org/commands/plugin/install/)) is one command. `wp plugin activate` ([doc](http://wp-cli.org/commands/plugin/activate/)) is another.
+
+WP-CLI comes with dozens of commands. But it's easier than it looks to create a custom WP-CLI command. Read the [commands cookbook](http://wp-cli.org/docs/commands-cookbook/) to learn more.
```
Comment (danielbachhuber) [inline README.md]: You can drop the "But" — there's no negation of the prior statement.

> It's easier than it looks to create a custom WP-CLI command.

**Comment by danielbachhuber** [comment]
> Ok, updated according to your comments. All pretty straightforward except take a look at the language under Support and the second demonstration of a command under Using.

Looking really good! Just a few more nitpick comments and then I think it's good to 🚢

**Comment by NateWr** [comment]
Ok, updated the PR with your changes. Sorry for the delay.

**Comment by danielbachhuber** [comment]
@NateWr Looks great! Thanks for your work on this.

**Pull request merged**